test/TestUtilities/Test.Utility/FileSystemBackedV3MockServer.cs

```diff
@@ -44,6 +46,18 @@ private void InitializeServer()
 
             Get.Add("/", request =>
             {
+                if (_isPrivateFeed)
+                {
+                    string authorization = request.Headers["Authorization"];
+                    if (authorization == null)
+                    {
+                        return new Action<HttpListenerResponse>(response =>
+                        {
+                            response.StatusCode = 401;
+                            response.AddHeader("WWW-Authenticate", "Basic");
+                        });
+                    }
+                }
```

Comment:
Considering that this PR pertains to a private feed requiring an authorization header, I have incorporated logic to ensure the code utilizes credential providers to obtain credentials.